lib/omniauth/strategies/magic_link.rb

```diff
@@ -15,6 +15,8 @@ class MagicLink
 
       def request_phase
         email = request[:email]
+        raise InvalidEmailError unless EmailValidator.new(email).valid?
```

Comment:
Ako sa toto sprava ked zadam zly email?

Reply:
@jsuchal je tam frontendova validacia, cize samotne emailove policko nedovoli submit prazdneho pola, alebo niecoho co nie je email. Commitol som handling na backende pomocou tej metody fail! 

Akurat co pozeram, ze argument metody fail! sa nikde nespracovava a vyrenderuje sa failure.html.erb, kde su hardcoded texty. 

V tomto pripade ale frontendova validacia kryje tu situaciu, cize pri logine by sa failure stranka nemala vlastne nikdy ani zobrazit.

Reply:
Super, dik. Tym padom potrebujeme vlastne zmeny v `app/views/sessions/new.html.erb` ?

Reply:
@eronisko Odstranil som celu tu logiku okolo @email_error, lebo to uz vlastne nerobime. Myslim ze zmeny v `app/views/sessions/new.html.erb` nepotrebujeme, lebo tam sa neredirectujeme. K volaniu toho naseho fail! ani nema ako dojst vdaka tej frontend validacii a keby doslo, ta metoda to redirectuje na `failure.html.erb`. 

Cize ma napada, skor na tej `failure.html.erb` mat nejaky mechanizmus zobrazenia spravnej chybovej hlasky podla aktualneho obsahu params[:message]. Napriklad helper metoda, ktora podla predaneho parametru vrati texty do tej stranky, co povies?

Reply:
Super, fajn. Za mna toto uplne staci. Bypassovanie tych front-end validacii je podla mna vymyslanie a take use casey podla mna nemusime teraz handlovat :)